mrmustard/physics/representations.py

```diff
@@ -310,3 +308,143 @@ def plot(
         ax.set_title(f"${title}$")
         plt.show(block=False)
         return fig, ax
+
+class Fock(Representation):
+    r"""This class is the Fock representation of a broad class of quantum states,
+    transformations, measurements, channels, etc.
+
+    The ansatz available in this representation is the multidimensional Tensor.
+
+    This class allows for operations of Fock tensors.
+
+    Examples:
+        .. code-block:: python
+            #TODO
+            A = math.astensor([[[1.0]]])  # 1x1x1
+            b = math.astensor([[0.0]])    # 1x1
+            c = math.astensor([0.9])      # 1
+            psi1 = Bargmann(A, b, c)
+            psi2 = Bargmann(A, b, c)
+            psi3 = 1.3 * psi1 - 2.1 * psi2  # linear combination
+            assert psi3.A.shape == (2, 1, 1)  # stacked along batch dimension
+            psi4 = psi1[0] @ psi2[0]  # contract wires 0 on each (inner product)
+            assert psi4.A.shape == (1,)  # A is 0x0 now (no wires left)
+            psi5 = psi1 & psi2  # outer product (tensor product)
+            rho = psi1.conj() & psi1   # outer product (this is now the density matrix)
+            assert rho.A.shape == (1, 2, 2)  # we have two wires now
+            assert np.allclose(rho.trace((0,), (1,)), np.abs(c)**2)
+    Args:
+
+    """
+
+    def __init__(
+        self,
+        array: Batch[Tensor]
+    ):
+        r"""Initializes the Fock representation.
+        Note that args needs to flag is batched or not, if it is non-batched, add the batch 1.
+
+        Args:
+            array: batch of the object array
+        """
+        self._contract_idxs: tuple[int, ...] = ()
+        self.ansatz = ArrayAnsatz(array=array)
+
+    def from_ansatz(self, ansatz: ArrayAnsatz, batched: bool) -> Fock:
+        r"""Returns a Fock object from an ansatz object."""
+        return self.__class__(ansatz.array)
+
+    @property
+    def array(self) -> Batch[Tensor]:
+        r"""
+        The array.
+        """
+        return self.ansatz.array
+
+    def conj(self):
+        r"""
+        The conjugate of this Fock object.
+        """
+        new = self.__class__(np.conj(self.array[1:]))
+        new._contract_idxs = self._contract_idxs
+        return new
+
+    def __getitem__(self, idx: int | tuple[int, ...]) -> Bargmann:
+        r"""Returns a copy of self with the given indices marked for contraction."""
+        idx = (idx,) if isinstance(idx, int) else idx
+        for i in idx:
+            if i >= self.ansatz.array.dim:
+                raise IndexError(
+                    f"Index {i} out of bounds for ansatz {self.ansatz.__class__.__qualname__} of dimension {self.ansatz.dim}."
+                )
+        new = self.__class__(self.array[idx])
```

Comment:
getitem here should be to set the values in `_contract_indices`, not to get a subset of array